src/components/custom/Product/ProductPart.tsx

```diff
@@ -15,20 +19,55 @@ interface ProductPartProps {
 
 export const ProductPart = ({ user, isAdmin }: ProductPartProps) => {
   const t = useTranslations("ProductPart");
-  const { userMemberships } = useUserMemberships(user.id);
-  const { purchases, total: totalToPay } = useUserPurchases(user.id);
+
+  const { toast } = useToast();
+  const [isLoading, setIsLoading] = useState(false);
+  const { memberships } = useUserMemberships(user.id);
+  const { purchases, total: totalToPay, refetch } = useUserPurchases(user.id);
   const { products: allProducts } = useProducts();
+
   const allConstraint = allProducts
     .map((product) => product?.product_constraints)
     .flat();
   const allConstraintIds = allConstraint?.map((constraint) => constraint?.id);
   const userAssociationsMembershipsIds = userMemberships.map(
     (membership) => membership.association_membership_id,
   );
+
+  const handleValidateAll = async () => {
+    setIsLoading(true);
+    try {
+      await Promise.all(
+        purchases.map((purchase) =>
+          onValidate(purchase, user, setIsLoading, refetch, toast),
+        ),
```

Comment:
A better way to do it would of course to have a backend endpoint that does all the validation at once. This should be a momentary behavior since it starts n call instead of 1 with a better suited backend.